addons/stock/tests/test_stock_return_picking.py

```diff
@@ -104,3 +105,33 @@ def test_return_picking_SN_pack(self):
         picking2.move_lines.move_line_ids.qty_done = 1
         picking2.button_validate()
         self.assertFalse(self.env['stock.quant']._gather(product_serial, customer_location, lot_id=serial1))
+
+    def test_return_incoming_picking(self):
+        partner = self.env['res.partner'].create({'name': 'Jean'})
+        picking_attributes = {
+            'picking_type_id': self.picking_type_in,
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location,
+            'partner_id': partner.id
+        }
+        picking_in = self.env['stock.picking'].create(picking_attributes)
+        self.MoveObj.create({
+            'name': self.UnitA.name,
+            'product_id': self.UnitA.id,
+            'product_uom_qty': 2,
+            'quantity_done': 2,
+            'product_uom': self.uom_unit.id,
+            'picking_id': picking_in.id,
+            'location_id': self.stock_location,
+            'location_dest_id': self.customer_location})
+        picking_in.button_validate()
+        # create a return picking
+        stock_return_picking_form = Form(self.env['stock.return.picking']
+            .with_context(active_ids=picking_in.ids, active_id=picking_in.ids[0],
+            active_model='stock.picking'))
+        stock_return_picking = stock_return_picking_form.save()
+        stock_return_picking.product_return_moves.quantity = 2.0
+        stock_return_picking_action = stock_return_picking.create_returns()
+        return_picking = self.env['stock.picking'].browse(stock_return_picking_action['res_id'])
+        return_picking.button_validate()
+        self.assertEqual(return_picking.move_lines[0].partner_id.id, picking_in.partner_id.id, "Stock moves of return picking should have partner_id similar of the original picking")
```

Comment:
```suggestion
        self.assertEqual(return_picking.move_lines[0].partner_id.id, picking_in.partner_id.id)
```
Assert msg should only be used to explain the expected value. Here the message does not provide any info